workers/unity/Packages/com.improbable.gdk.core/Worker/ConnectionHandlers/MockConnectionHandler.cs

```diff
@@ -0,0 +1,90 @@
+using Improbable.Worker.CInterop;
+
+namespace Improbable.Gdk.Core
+{
+    public class MockConnectionHandler : IConnectionHandler
+    {
+        private uint updateId;
+
+        private readonly ViewDiff diff = new ViewDiff();
+
+        public void ClearDiff()
+        {
+            diff.Clear();
+        }
+
+        public void CreateEntity(long entityId, EntityTemplate template)
+        {
+            var handler = new EntityTemplateDynamicHandler(template, entityId, diff);
+            DynamicConverter.ForEachComponent(handler);
+        }
+
+        public void ChangeAuthority(long entityId, uint componentId, Authority newAuthority)
+        {
+            diff.SetAuthority(entityId, componentId, newAuthority);
+        }
+
+        public void UpdateComponent<T>(long entityId, uint componentId, T update) where T : ISpatialComponentUpdate
+        {
+            diff.AddComponentUpdate(update, entityId, componentId, updateId++);
+        }
+
+        public void AddEvent<T>(long entityId, uint componentId, T ev) where T : IEvent
+        {
+            diff.AddEvent(ev, entityId, componentId, updateId++);
+        }
+
+        // TODO: Commands
+
+        #region IConnectionHandler implementation
+
+        public ViewDiff GetMessagesReceived()
+        {
+            return diff;
```

Comment:
This looks like you edit the diff as you are reading it

Reply:
Its up to the caller to ensure things happen in the correct order.

Alternatively we can have two diffs that we swap around